connectivity/http/http.c

```diff
@@ -452,6 +394,10 @@ static int ta_http_handler(void *cls, struct MHD_Connection *connection, const c
 
   /* decide which API function should be called */
   req_ret = ta_http_process_request(api, url, http_req->request, &response_buf, options);
+  if (req_ret == SC_CCLIENT_JSON_CREATE) {
+      ta_log_error("%s\n", "Failed to process http request.");
+      goto cleanup;
+  }
```

Comment:
If this happened, the client won't receive a response, right?
And since we already log the message in `ta_http_process_request`, I think it's nice to response `internal service error` to client.

so the workflow in `ta_http_process_request` when JSON creation is failed:
1. log error in TA, and specify this happened in http response creation 
2. set `ret` to `SC_CCLIENT_JSON_CREATE`
3. the switch case goes to `Internal service error`
4. return

Then the check here can be removed, and let MHD do the rest.

What do you think, @HowJMay @afcidk @YingHan-Chen ?

Reply:
Do you mean the switch case in [common.c:63](https://github.com/DLTcollab/tangle-accelerator/pull/484/files#diff-57d73caeae12f79fe4979aab5ae662feR63)? 
I've noticed this issue when writing this part of code, but wondering how could we send the response if we cannot create the response? 

Alternatively, could we create and keep the cJSON objects when TA starts instead of creating and deleting everytime we need to set response content?

Reply:
Ah, yeah! You're totally right, I'm being silly here :stuck_out_tongue: 

You mean global objects?
But TA works concurrently, it handles multiple requests at a same time.
do you have any idea?  :thinking:

Reply:
With only read to the global objects, race condition won't happen. Does it seem reasonable to define the objects as constant?